Sources/DistributedActors/Cluster/Reception/OperationLogDistributedReceptionist.swift

```diff
@@ -410,25 +410,30 @@ extension OpLogDistributedReceptionist: LifecycleWatch {
 
 extension OpLogDistributedReceptionist {
     func ensureDelayedListingFlush(of key: AnyDistributedReceptionKey) {
-        let timerKey = self.flushTimerKey(key)
-
         if self.storage.registrations(forKey: key)?.isEmpty ?? true {
             self.log.debug("notify now, no need to schedule delayed flush")
             self.notifySubscribers(of: key)
             return // no need to schedule, there are no registered actors at all, we eagerly emit this info
         }
 
-        guard !self.timers.exists(key: timerKey) else {
+        let timerTaskKey = key.hashValue
+        guard self.flushTimerTasks[timerTaskKey] == nil else {
             self.log.debug("timer exists")
             return // timer exists nothing to do
         }
 
         // TODO: also flush when a key has seen e.g. 100 changes?
         let flushDelay = actorSystem.settings.receptionist.listingFlushDelay
-        // timers.startSingle(key: timerKey, message: _ReceptionistDelayedListingFlushTick(key: key), delay: flushDelay)
         self.log.debug("schedule delayed flush")
-        self.timers.startSingle(key: timerKey, delay: flushDelay) {
-            self.onDelayedListingFlushTick(key: key)
+        self.flushTimerTasks[timerTaskKey] = Task {
+            for await _ in AsyncTimerSequence(interval: flushDelay, clock: ContinuousClock()) {
+                self.onDelayedListingFlushTick(key: key)
+                // Single-shot; cancel task immediately after it has been fired
+                if let timerTask = self.flushTimerTasks.removeValue(forKey: timerTaskKey) {
+                    timerTask.cancel()
+                }
+                break
```

Comment:
That's also a single timer, so we can do 

```
self.flushTimerTasks[timerTaskKey] = Task { 
  defer { self.flushTimerTasks.removeValue(forKey: timerTaskKey) } 
  try await Task.sleep(until: .now + flushDelay, clock: .continuous) 
}
```

hmm, the cancel seems un-necessary then as well; since we'll run to completion, but we do want to keep the `removeValue` 👍